### ISSUE TITLE
Type alias getting "DeclarationTypeChanged"

### ISSUE THREAD

**Issue by jpillora** [smola/gocompat#22]
While technically true, Declaration Type did Change. It's a compatible change. Is this expected behaviour:

https://github.com/jpillora/gocompat/commit/e9400c2bb9ad16a65820a1bcc082617169f5f59b

```
"github.com/jpillora/gocompat".Add DeclarationTypeChanged
"github.com/jpillora/gocompat".Num DeclarationTypeChanged
found backwards incompatible changes
```

**Comment by jpillora** [comment]
Maybe there's compatibility issues from going from `func` to `var`? However the type alias should be fine right? My understanding is the purpose of type aliases was to perform exactly this kind of change

**Comment by smola** [comment]
@jpillora This is currently the expected behavior. Changing an existing type to an alias is not backwards compatible. For example:

- Reflection: `reflect.TypeOf` on `Num` would return `gocompat.Num` before the change, but it will be `newmath.Num` after it.
- Equality: Before the change, `gocompat.Num` and `newmath.Num` values would not be equal. After the change, they are. This can break code that was previously dealing with both types and is particularly relevant if they come from different modules.

That being said, I think it would make sense to have more fine-grained output here and signaling when the only change in the declaration is the presence of an alias or the change of an alias target.

**Comment by jpillora** [comment]
RE: Reflection: true - though this is a runtime compatibility issue, similar to introducing a logical error

RE: Equality: before the change `newmath` didn't exist. Are you refering to a case where `newmath.Num` existed, but referred to a different type?

**Comment by jpillora** [comment]
Basically, what I'm looking for is "will old programs compile with these changes?" and then any runtime issues can be resolved after that

**Comment by smola** [comment]
> Equality: before the change newmath didn't exist.

Yep. I'll check if there is a reasonable way to check that. It can probably done if both types belong to the same module.

> Basically, what I'm looking for is "will old programs compile with these changes?"

Ok. Sounds reasonable.
